assets/js/admin.js

```diff
@@ -637,3 +626,7 @@ jQuery(document).on('sonata-admin-append-form-element', (event) => {
   Admin.setup_icheck(event.target);
   Admin.setup_collection_counter(event.target);
 });
+
+jQuery('select.per-page').on('change', (event) => {
+  event.target.dispatchEvent(new Event('change'));
```

Comment:
Select2 not dispatch native `change` event only for jquery, in branch 5.x will be removed, because select will be native.